Features/Contacts/Sources/Scenes/AddContactAddressScene.swift

```diff
@@ -0,0 +1,128 @@
+// Copyright (c). Gem Wallet. All rights reserved.
+
+import SwiftUICore
+import SwiftUI
+import PrimitivesComponents
+import Components
+import Primitives
+import Style
+import Localization
+import NameResolver
+
+public struct AddContactAddressScene: View {
+    
+    @State private var model: AddContactAddressViewModel
+    @State private var presentingErrorMessage: String?
+
+    @FocusState private var focusedField: Field?
+    enum Field: Int, Hashable {
+        case address
+        case name
+        case description
+    }
+    
+    public init(model: AddContactAddressViewModel) {
+        _model = State(initialValue: model)
+    }
+    
+    @ViewBuilder
+    public var body: some View {
+        VStack {
+            List {
+                Section {
+                    if let chain = model.input.chain.value {
+                        NavigationLink(value: Scenes.NetworksSelector()) {
+                            ChainView(model: ChainViewModel(chain: chain))
+                        }
+                    } else {
+                        NavigationLink(value: Scenes.NetworksSelector()) {
+                            ListItemView(title: "Select chain")
+                        }
+                    }
+                    if let chain = model.input.chain.value {
+                        FloatTextField(model.addressTextFieldTitle, text: $model.projectedValue.input.address.value, allowClean: false) {
+                            HStack(spacing: .large/2) {
+                                NameRecordView(
+                                    model: NameRecordViewModel(chain: chain),
+                                    state: $model.nameResolveState,
+                                    address: $model.input.address.value
+                                )
+                                ListButton(image: Images.System.paste) {
+                                    onSelectPaste()
+                                }
+                                ListButton(image: Images.System.qrCode) {
+                                    onSelectScan()
+                                }
+                            }
+                        }
+                        .focused($focusedField, equals: .address)
+                        .keyboardType(.alphabet)
+                        .textInputAutocapitalization(.never)
+                        .autocorrectionDisabled()
+                        .truncationMode(.middle)
+                    }
+                    if model.input.chain.value?.isMemoSupported == true {
```

Comment:
```suggestion
                    if model.showMemo {
```

`input` should not be exposed in view model